integrations/maltiverse.py

```diff
@@ -0,0 +1,599 @@
+# Created by Maltiverse.
+#
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+"""
+What is Maltiverse?
+###################
+
+Maltiverse works as a broker for Threat intelligence sources that are
+aggregated from more than a hundred different Public, Private and Community
+sources. Once the data is ingested, the IoC Scoring Algorithm applies a
+qualitative classification to the IoC that changes. Finally this data can
+be queried in a Threat Intelligence feed that can be delivered to your
+Firewalls, SOAR, SIEM, EDR or any other technology.
+
+What does this integration?
+###########################
+
+This integration enrichs any alert generated by Wazuh via the Maltiverse API,
+inserting new fields in case of match and following the threat taxonomy of the
+ECS standard (Elastic Common Squema).
+
+https://www.elastic.co/guide/en/ecs/current/ecs-threat.html
+
+Ipv4, Domain names, Urls and MD5/SHA1 checksums are checked in Maltiverse
+platform in order to enrich the original alert with threat Intel information
+
+Installation Guide
+##################
+
+Add this to the ossec.conf file, inside <ossec_config></ossec_config> block:
+
+    <integration>
+        <name>maltiverse</name>
+        <hook_url>https://api.maltiverse.com</hook_url>
+        <api_key><YOUR_MALTIVERSE_AUTH_TOKEN></api_key>
+        <alert_format>json</alert_format>
+    </integration>
+
+And restart Wazuh Manager:
+
+    /etc/init.d/wazuh-manager restart
+
+"""
+
+import json
+import hashlib
+import ipaddress
+import os
+from socket import socket, AF_UNIX, SOCK_DGRAM
+import sys
+import time
+from urllib.parse import urlsplit
+
+try:
+    import requests
+except Exception as e:
+    print("No module 'requests' found. Install: pip install requests")
+    sys.exit(1)
+
+# Global vars
+debug_enabled: bool = False
+pwd: str = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+json_alert: dict = {}
+now: str = time.strftime("%a %b %d %H:%M:%S %Z %Y")
+
+# Set paths
+LOG_FILE: str = f"{pwd}/logs/integrations.log"
+SOCKET_ADDR: str = f"{pwd}/queue/sockets/queue"
+
+
+class Maltiverse:
+    """This class is a simplification of maltiverse pypi package."""
+
+    def __init__(
+        self, endpoint: str = "https://api.maltiverse.com", auth_token: str = None
+    ):
+        """Initialize the Maltiverse class.
+
+        Parameters
+        ----------
+        endpoint : str, optional (default: "https://api.maltiverse.com")
+            The API endpoint URL.
+        auth_token : str, optional
+            The authentication token for the API.
+        """
+        self.endpoint = endpoint
+        self.auth_token = auth_token
+        self.headers = {
+            "Accept": "application/json",
+            "Authorization": f"Bearer {self.auth_token}",
+        }
```

Comment:
I suggest instantiating a [Session](https://docs.python-requests.org/en/latest/user/advanced/#session-objects) with the headers and using them in the successive requests.